agent/xds/clusters.go

```diff
@@ -130,7 +130,7 @@ func (s *ResourceGenerator) clustersFromSnapshotConnectProxy(cfgSnap *proxycfg.C
 	// NOTE: Any time we skip an upstream below we MUST also skip that same
 	// upstream in endpoints.go so that the sets of endpoints generated matches
 	// the sets of clusters.
-	for _, uid := range cfgSnap.ConnectProxy.PeeredUpstreamIDs() {
+	for _, uid := range cfgSnap.ConnectProxy.PeeredUpstreamIDsFromConfig() {
```

Comment:
Wasn't sure whether I should: 1) split this function (current decision); or 2) make `PeeredUpstreamIDs` return implicit+explicit upstreamIDs and do a `continue` in the loop if we detect that `upstreamCfg == nil`